### PR TITLE
Editor - Resizing elements inside page rows

### PR DESCRIPTION
**Pull request by apedroferreira**
Related issues: https://github.com/mui/mui-toolpad/issues/101 and https://github.com/mui/mui-toolpad/issues/359

Add ability to resize node elements inside page rows, in the editor:

https://user-images.githubusercontent.com/10789765/179825987-01df4485-717d-4085-9742-ac2580d8a592.mov

- allow dragging sides of elements to resize them
- show grid overlay while dragging - snap to grid columns to make it easier for users to build a good-looking layout
- show resize preview while dragging - changing the styles in the DOM directly, using a CSS transform scale
- control grid layout inside rows through a prop set directly in the runtime - use the `layout` namespace in page row children to control the value of that prop
- includes some necessary small adjustments to some editor components too

closes https://github.com/mui/mui-toolpad/issues/681

### CODE REVIEW

**Comment by apedroferreira** [comment]
@Janpot Resizing to the outside kind of works (need to fix resizing to the inside), but performance is pretty bad at the moment... (will refactor things a bit later too, I'm mostly trying to make it work now)
Are we doing any kind of debouncing/delay when updating the DOM os something? Just need to know, otherwise I can try to find out how to make the performance smoother.

File: packages/toolpad-app/src/components/AppEditor/PageEditor/RenderPanel.tsx [hunk 1]
```diff
@@ -757,11 +813,119 @@ export default function RenderPanel({ className }: RenderPanelProps) {
...
   }, [nodesInfo, pageNodes]);
 
+  const updatePageRowLayout = React.useCallback(
+    (pageRowNode: appDom.ElementNode) => {
+      const children = appDom.getChildNodes(dom, pageRowNode).children;
+      const layoutColumnSizes = children
+        .map((child) => child.layout?.columnSize || appDom.createConst(0))
+        .map((constAttribute) => constAttribute.value);
+
+      const totalLayoutColumnsSize = layoutColumnSizes.reduce((acc, size) => acc + size, 0);
```
Comment (apedroferreira) [inline packages/toolpad-app/src/components/AppEditor/PageEditor/RenderPanel.tsx:823]: will extract this column size normalization logic later

**Comment by apedroferreira** [comment]
i'll also probably remove the edge drag areas outside of the `NodeHud`, doesn't make sense for them to be there.

File: packages/toolpad-app/src/toolpad/AppEditor/PageEditor/RenderPanel/RenderPanel.tsx [hunk 2]
```diff
@@ -1058,6 +1142,16 @@ export default function RenderPanel({ className }: RenderPanelProps) {
                 lastContainerChild.parentProp,
                 parent.parentIndex,
               );
+
+              if (isPageColumn(parent)) {
```
Comment (apedroferreira) [inline packages/toolpad-app/src/toolpad/AppEditor/PageEditor/RenderPanel/RenderPanel.tsx:1146]: i'll probably improve this function later to not have to repeat as much logic- should be done separately though as it relates to the whole editor and not just resizing

**Review by Janpot — APPROVED** [review]
👍  lgtm

**Pull request merged**